### PR TITLE
support multi-module modules.yaml

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: koji/collection-mash-split.py [hunk 1]
```diff
@@ -160,13 +160,14 @@ def inject_modulemd_yml(self, path, modulemd_yaml, modulemd_version=0, modulemd_
         cmd += " --queryformat='%{name}-%{epochnum}:%{version}-%{release}.%{arch}\n'"
         self.logger.debug("running %s" % cmd)
         status, output = kobo.shortcuts.run(cmd)
-        modules = yaml.safe_load(modulemd_yaml)
-        modules['data']['artifacts'] = {'rpms': output.splitlines()}
-        modules['data']['version'] = modulemd_version
-        modules['data']['context'] = modulemd_context
+        modules = yaml.safe_load_all(modulemd_yaml)
+        for module in modules:
+            modules['data']['artifacts'] = {'rpms': output.splitlines()}
```
Comment (evgeni) [inline koji/collection-mash-split.py:165]: this should only happen for the "main" module

File: koji/collection-mash-split.py [hunk 2]
```diff
@@ -160,13 +161,15 @@ def inject_modulemd_yml(self, path, modulemd_yaml, modulemd_version=0, modulemd_
...
-        modules = yaml.safe_load(modulemd_yaml)
-        modules['data']['artifacts'] = {'rpms': output.splitlines()}
-        modules['data']['version'] = modulemd_version
-        modules['data']['context'] = modulemd_context
+        modules = list(yaml.safe_load_all(modulemd_yaml))
+        for module in modules:
+            modules['data']['version'] = modulemd_version
+            modules['data']['context'] = modulemd_context
+            if modules['data']['name'] == collection:
+                modules['data']['artifacts'] = {'rpms': output.splitlines()}
```
Comment (ekohl) [inline koji/collection-mash-split.py:169]: Shouldn't this be `module` now?
```suggestion
            module['data']['version'] = modulemd_version
            module['data']['context'] = modulemd_context
            if module['data']['name'] == collection:
                module['data']['artifacts'] = {'rpms': output.splitlines()}
```

Comment (evgeni) [reply koji/collection-mash-split.py]: of course!

**Review by ekohl — APPROVED** [review]

**Comment by evgeni** [comment]
(deployed)

**Pull request merged**